components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/loader/shared/SharedURLClassLoader.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.webapp.mgt.loader.shared;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.webapp.mgt.loader.WebappClassloadingContext;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+
+/**
+ * This classloader will be the parent of the CarbonWebappClassLoader class loader.
+ * It will loop through the shared environment class loaders to load the resource.
+ */
+public class SharedURLClassLoader extends URLClassLoader {
```

Comment:
Perhaps this class can be made non-public, as it will only be created via the Factory. Hence there may not need to give public access.

Reply:
Changed to non-public